runtime/runtime/src/actions.rs

```diff
@@ -438,6 +442,18 @@ pub(crate) fn action_deploy_contract(
         })?;
     account.code_hash = code.get_hash();
     set_code(state_update, account_id.clone(), &code);
+    if let Some(cache) = cache {
+        // We only precompile contract for Wasmer VM at the moment.
+        if let Some(err) =
+            precompile(code.get_code(), &code.get_hash(), vm_config, cache.deref(), VMKind::Wasmer)
+        {
+            return Err(StorageError::StorageInconsistentState(format!(
+                "Cannot precompile contract {}: {:?}",
+                code.get_hash().to_string(),
```

Comment:
```suggestion
                code.get_hash(),
```

`x.to_string` is, by definition, `format!("{}", x)`, so `format!("{}", x.to_string())` is a tautology.

Reply:
Thanks, C still haunts my brain.